open-apparel-registry/open-apparel-registry
# Better accommodate outside contributors

## Overview

These changes were made in an effort to better accommodate outside contributors:

- Because `.env.frontend` was only used by the legacy NodeJS API, consolidate everything into a single `.env` file. I also created a new `.env` file on S3
- Update `scripts/bootstrap` to generate a stub `.env` file, on the first run, if the file on S3 is inaccessible
- Update the README file so that non-Azaveans can quickly setup an instance of OAR to contribute to the project


These changes are additional:

- Update Ansible to the 2.8.x series
- Update the Compose file format to 2.4

Resolves #14 

### Checklist


## Testing Instructions

Verify that, without an AWS profile, you can follow the instructions in the README to spin up and interact with an instance of OAR.

---

See that, without an AWS profile, `scripts/bootstrap` will stub out a new `.env` file:

```bash
$ rm .env
$ export AWS_PROFILE=asdfghjkl
$ ./scripts/bootstrap
...
botocore.exceptions.ProfileNotFound: The config profile (asdfghjkl) could not be found
Updated environment variables file from .env.sample
~/projects/open-apparel-registry
```

See that, without an AWS profile, `scripts/bootstrap` will not overwrite an existing `.env` file:

```bash
$ ./scripts/bootstrap
...
botocore.exceptions.ProfileNotFound: The config profile (asdfghjkl) could not be found
~/projects/open-apparel-registry
```

See that, with a AWS profile, `scripts/bootstrap` will always overwrite with the contents of the file in S3:

```bash
$ export AWS_PROFILE=open-apparel-registry
$ ./scripts/bootstrap
~/projects/open-apparel-registry ~/projects/open-apparel-registry
download: s3://openapparelregistry-development-config-eu-west-1/.env to ./.env
Updated environment variables file from openapparelregistry-development-config-eu-west-1
~/projects/open-apparel-registry
```
